src/chain/shared_state.rs

```diff
@@ -156,6 +157,32 @@ impl SharedState {
                 .collect()
         }
     }
+
+    /// Updates the entry in `their_keys` for `prefix` to the latest known key; if a split
+    /// occurred in the meantime, the keys for sections covering the rest of the address space are
+    /// initialised to the old key that was stored for their common ancestor
+    pub fn update_their_keys(&mut self, prefix: Prefix<XorName>, key: BlsPublicKey) {
+        if let Some(&pfx) = self
+            .their_keys
+            .keys()
+            .find(|pfx| pfx.is_compatible(&prefix))
+        {
+            let old_key = unwrap!(self.their_keys.remove(&pfx));
+            let old_pfx_sibling = pfx.sibling();
+            let mut current_pfx = prefix.sibling();
+            while !self.their_keys.contains_key(&current_pfx) && current_pfx != old_pfx_sibling {
+                let _ = self.their_keys.insert(current_pfx, old_key.clone());
+                current_pfx = current_pfx.popped().sibling();
+            }
+        }
+        let _ = self.their_keys.insert(prefix, key);
+    }
+
+    #[cfg(test)]
+    /// Returns the relevant entry in their_keys if it exists
```

Comment:
this function actually `gives out their_keys if have` ? 
the comment of `returns the relevant entry` indicates it shall be something like `get_their_keys(prefix)` ?

Reply:
Right, the old version of the function did that, but I forgot to change the comment. Good catch :+1: